### DISCUSSION TITLE
How do i "run go build in root directory of project"?

### DISCUSSION THREAD

**Discussion by Rosey723** [community/community#129555] [Programming Help]
### Select Topic Area

Question

### Body

I am trying to install https://github.com/21Bruce/resolved-bot?tab=readme-ov-file#readme.
I finally got go programming language running. I hit Code and downloaded as a zip and downloaded on GitHub Desktop. I can't figure out how to "run go build". When i try typing in go build in cmd i get "go: go.mod file not found in current directory or any parent directory; see 'go help modules'". I am new to Github so it's probably that but i need help installing this. Can someone tell me what i'm supposed to be doing to get this installed?

**Comment by verduscos** [comment]
Based on the error you encountered, it seems you're trying to execute the command in a directory without a go.mod file. This indicates you're not in the project's root directory, as specified in the repository's readme. Navigate to your cloned project's directory using cd, ensuring you're in the directory containing go.mod file, then try running go build.

**Reply by Rosey723** [reply]
Do you know how i can "navigate to your cloned project's directory using cd"?

**Reply by verduscos** [reply]
You mentioned you downloaded the zip file of the repo you're trying to work with. You need to unzip those files somewhere on your computer. In cmd you need to navigate to that directory using the [cd](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/cd) before running that go command.

**Reply by Rosey723** [reply]
Could you tell me what code i need to type in. I have it in my downloads folder.

**Reply by verduscos** [reply]
For example, if you're at C:\Users\Username> and your downloads folder location is at C:\User\Username\downloads, then you'd run cd downloads. You'll just need to make sure of the exact path to the location of that folder.

**Reply by ebndev** [reply]
Hi @verduscos, thanks for taking the time to answer!

@Rosey723 if the response above helped, please feel encouraged to mark it as the answer. If not, let us know if you have further questions or post the solution if you've resolved it on your own. 😄
